src/main/scala/viper/server/vsi/VerificationServerInterface.scala

```diff
@@ -0,0 +1,310 @@
+        sender ! startJob(task, queue, publisher)
+      case msg =>
+        throw new Exception("Main Actor: unexpected message received: " + msg)
+    }
+
+    private def startJob(task: Thread, queue: SourceQueueWithComplete[Envelope], publisher: Publisher[Envelope]): JobHandle = {
+      _verificationTask = task
+      _verificationTask.start()
+      println(s"Starting job #$id...")
+      JobHandle(self, queue, publisher)
+    }
+  }
+
+
+  // --- Actor: MessageActor ---
+
+  object MessageActor {
+    def props(jid: Int, queue: SourceQueueWithComplete[Envelope]): Props = Props(new MessageActor(jid, queue))
+  }
+
+  class MessageActor(jid: Int, queue: SourceQueueWithComplete[Envelope]) extends Actor {
+
+    override def receive: PartialFunction[Any, Unit] = {
+      case TaskProtocol.ClientRequest =>
+      case TaskProtocol.ServerReport(msg) =>
+        val offer_status = queue.offer(msg)
+        sender() ! offer_status
+      case TaskProtocol.FinalServerReport(success) =>
+        queue.complete()
+        if ( success )
+          println(s"Job #$jid has been completed successfully.")
+        else
+          println(s"Job #$jid has been completed ERRONEOUSLY.")
+        self ! PoisonPill
+      case _ =>
+    }
+  }
+
+  protected def start()
+
+  /** This method starts an individual verification process.
+    *
+    * As such, it accepts an instance of a VerificationTask.
+    */
+  protected def initializeVerificationProcess(task:VerificationTask): VerificationJobHandler = {
+    if (jobs.newJobsAllowed) {
+      def createJob(new_jid: Int): Future[JobHandle] = {
+        implicit val askTimeout: Timeout = Timeout(5000 milliseconds)
+        val job_actor = system.actorOf(JobActor.props(new_jid), s"job_actor_$new_jid")
+        val (queue, publisher) = Source.queue[Envelope](10000, OverflowStrategy.backpressure)
+                                       .toMat(Sink.asPublisher(false))(Keep.both)
+                                       .run()
+        val message_actor = system.actorOf(MessageActor.props(new_jid, queue), s"queue_actor_$new_jid")
+        task.setQueueActor(message_actor)
+        val task_with_actor = new Thread(task)
+        val answer = job_actor ? VerificationProtocol.Verify(task_with_actor, queue, publisher)
+        val new_job_handle: Future[JobHandle] = answer.mapTo[JobHandle]
+        new_job_handle
+      }
+      val (id, _) = jobs.bookNewJob(createJob)
+      VerificationJobHandler(id)
+    } else {
+      println(s"the maximum number of active verification jobs are currently running ($jobs.MAX_ACTIVE_JOBS).")
+      VerificationJobHandler(-1) // Not able to create a new JobHandle
+    }
+  }
+
+  protected def stop(): Unit = {
+    getInterruptFutureList() onComplete {
+      case Success(_) =>
+        _termActor ! Terminator.Exit
+        println(s"shutting down...")
+      case Failure(err_msg) =>
+        println(s"Interrupting one of the verification threads timed out: $err_msg")
```

Comment:
This should go to the logs, not necessarily to STDOUT.

Reply:
This is "old" code. A few weeks back we decided to keep old print statements as they were. Let me know if you still want me to change it.